### DISCUSSION TITLE
Load Balancer Rule Priority

### DISCUSSION THREAD

**Discussion by tim-xergy** [aws/copilot-cli#5623] [Q&A]
Hello,

I have an application that runs 2 services (web and api). The Web routes are less specific than the API routes. 

I had manually set the rule priorites so that requests check if they match the API rule set first as they are more specific (use a subdomain) and if they don't match they move on to the more generic web service. 

Unfortunately yesterday I had to update copilot cli int he pipeline due to an error caused by having custom rules in the load balancer.  Now the web rule priorities is set to be a higher priority than the API. 

Is it safe for me to re-priorities these in the console and is there a way to manually set the rule priorities in either the environment or service manifest?

**Comment by KollaAdithya** [comment]
Hello @tim-xergy !

That's a good question. I do not recommend you to manually modify the rule priorities which will cause `rule conflicts` when you add some more sub domains to the service.

Unfortunately Copilot by default assigns rule priorities  over [here](https://github.com/aws/copilot-cli/blob/mainline/cf-custom-resources/lib/alb-rule-priority-generator.js#L165:L191) in our custom resource for each domain that a copilot service uses.
There is no workaround as of now I could think for this issue.

This will be a feature request so that Copilot needs to expose an additional fields in the manifest [service/environment] to set custom priorities by the user.

**Comment by KollaAdithya** [comment]
Hello @tim-xergy 

Earlier I was wrong, there is a workaround using  [yaml patch overrides](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) to replace the Priorities assigned by Copilot

```
- op: replace
  path: /Resources/HTTPListenerRule/Properties/Priority
  value: 50000
- op: replace
  path: /Resources/HTTPListenerRuleWithDomain/Properties/Priority
  value: 49999
- op: replace
  path: /Resources/HTTPListenerRule1/Properties/Priority
  value: 1
- op: replace
  path: /Resources/HTTPListenerRuleWithDomain1/Properties/Priority
  value: 2
  ```

**Comment by tim-xergy** [comment]
@KollaAdithya Thanks for getting back to me! I only just noticed the notifications. I'll take a look into this thanks :)
